yt_dlp/extractor/niconicochannelplus.py

```diff
@@ -93,49 +166,51 @@ class NiconicoChannelPlusIE(NiconicoChannelPlusBaseIE):
             'tags': [],
             'upload_date': '20221021',
         },
-        'params': {
-            'skip_download': True,
-        },
+        'skip': 'subscriber only',
     }]
 
-    def _real_extract(self, url):
-        content_code, channel_id = self._match_valid_url(url).group('code', 'channel')
-        fanclub_site_id = self._find_fanclub_site_id(channel_id)
+    @staticmethod
+    def _match_video_id(url):
+        return re.search(r'/(?:video|audio|live)/(?P<id>sm\w+)', urllib.parse.urlparse(url).path)
 
-        data_json = self._call_api(
-            f'video_pages/{content_code}', item_id=content_code, headers={'fc_use_device': 'null'},
-            note='Fetching video page info', errnote='Unable to fetch video page info',
-        )['data']['video_page']
+    @classmethod
+    def suitable(cls, url):
+        try:
+            return super().suitable(url) or (
+                urllib.parse.urlparse(url).netloc in _SUITABLE_NICOCHANNEL_PLUS_DOMAINS
+                and cls._match_video_id(url))
+        except NameError:
+            # fallback for lazy extractor
+            return super().suitable(url)
```

Comment:
May revert to https://github.com/yt-dlp/yt-dlp/pull/11787/commits/941c50b69946093e73f92cb3f598b62ec1b082ff if this is too aggresive?

Reply:
Can you fix?

Reply:
Can you fix?